providers/hexonet/records.go

```diff
@@ -0,0 +1,277 @@
+package hexonet
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/StackExchange/dnscontrol/models"
+	"github.com/StackExchange/dnscontrol/providers/diff"
+)
+
+// HXRecord covers an individual DNS resource record.
+type HXRecord struct {
+	// Raw api value of that RR
+	Raw string
+	// DomainName is the zone that the record belongs to.
+	DomainName string
+	// Host is the hostname relative to the zone: e.g. for a record for blog.example.org, domain would be "example.org" and host would be "blog".
+	// An apex record would be specified by either an empty host "" or "@".
+	// A SRV record would be specified by "_{service}._{protocal}.{host}": e.g. "_sip._tcp.phone" for _sip._tcp.phone.example.org.
+	Host string
+	// FQDN is the Fully Qualified Domain Name. It is the combination of the host and the domain name. It always ends in a ".". FQDN is ignored in CreateRecord, specify via the Host field instead.
+	Fqdn string
+	// Type is one of the following: A, AAAA, ANAME, CNAME, MX, NS, SRV, or TXT.
+	Type string
+	// Answer is either the IP address for A or AAAA records; the target for ANAME, CNAME, MX, or NS records; the text for TXT records.
+	// For SRV records, answer has the following format: "{weight} {port} {target}" e.g. "1 5061 sip.example.org".
+	Answer string
+	// TTL is the time this record can be cached for in seconds.
+	TTL uint32
+	// Priority is only required for MX and SRV records, it is ignored for all others.
+	Priority uint32
+}
+
+// GetDomainCorrections gathers correctios that would bring n to match dc.
+func (n *HXClient) GetDomainCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
+	dc.Punycode()
+	records, err := n.getRecords(dc.Name)
+	if err != nil {
+		return nil, err
+	}
+	actual := make([]*models.RecordConfig, len(records))
+	for i, r := range records {
+		actual[i] = toRecord(r, dc.Name)
+	}
+
+	for _, rec := range dc.Records {
+		if rec.Type == "ALIAS" {
+			rec.Type = "CNAME" // TODO as we fully support ALIAS through our XDNS system
```

Comment:
Are these just plain cnames, or do your nameservers do some additional flattening of cnames? We use cnames for cloudflare, for example, in place of alias, because they transparently flatten them and serve as A records. Does hexonet do something like that? If not, then we should not claim to support alias records as they are semantically different from cnames.

Reply:
You were right, I've discussed this with our CTO. Falling back to CNAME is for sure wrong.
For ther 1st release, I've decided to leave out ALIAS support.
Our system supports ALIAS RR in correct way, but it creates in addition A Records or AAAA Records (copied from target zone) which will get updated from time to time.
So creating a ALIAS RR automatically comes with further assigned records. Not sure how dnscontrol can handle this. Is it possible to apply such additional and automatically created RR also to customer's dnsconfig file so that this stays transparent?

The easiest and working way would be to simply filter these A or AAAA RR out that are bound to that ALIAS RR, but I've to ensure that I can do this without running into further issues.

Reply:
Are you saying that if I make an ALIAS record, your system will create the underlying A/AAAA records, and actually return them form future api requests in addition to the original ALIAS record? That seems... odd. My naive expectation would probably be that those records get created only on the nameserver side of things, and remain hidden to the api consumer if possible. If they are indeed returned, then yes, they would probably need to be filtered out if identifiable in order to eliminate infinite update loop scenarios if running dnscontrol multiple times. We hope we can establish enough consistency so that if you ever run 'dnscontrol push' twice in a row, you should expect zero corrections on the second run.

Reply:
Looks like I was faced with an bug in our API. These A RRs shouldn't get returned. I will work on ALIAS feature in the next days. But as pointed out: this doesnt block 1st release.

Reply:
Launch of ALIAS is postponed for next weeks, no ETA for that backend system bugfix. So won't be part of the initial release. All parts of our module do already consider the change accordingly - no support for ALIAS. We plan this for the next release.

Thanks guys